consensus/hotstuff/validator/validator_test.go

```diff
@@ -537,11 +641,173 @@ func (qs *QCSuite) TestQCSignatureInvalidFormat() {
 }
+
+func TestValidateTC(t *testing.T) {
+	suite.Run(t, new(TCSuite))
+}
+
+type TCSuite struct {
+	suite.Suite
+	participants flow.IdentityList
+	signers      flow.IdentityList
+	indices      []byte
+	block        *model.Block
+	tc           *flow.TimeoutCertificate
+	committee    *mocks.DynamicCommittee
+	verifier     *mocks.Verifier
+	validator    *Validator
+}
+
+func (s *TCSuite) SetupTest() {
+
+	// create a list of 10 nodes with 1-weight each
+	s.participants = unittest.IdentityListFixture(10,
+		unittest.WithRole(flow.RoleConsensus),
+		unittest.WithWeight(1),
+	)
+
+	// signers are a qualified majority at 7
+	s.signers = s.participants[:7]
+
+	var err error
+	s.indices, err = signature.EncodeSignersToIndices(s.participants.NodeIDs(), s.signers.NodeIDs())
+	require.NoError(s.T(), err)
+
+	rand.Seed(time.Now().UnixNano())
+	view := uint64(int(rand.Uint32()) + len(s.participants))
+
+	highQCViews := make([]uint64, 0, len(s.signers))
+	for i := range s.signers {
+		highQCViews = append(highQCViews, view-uint64(i)-1)
+	}
+
+	rand.Shuffle(len(highQCViews), func(i, j int) {
+		highQCViews[i], highQCViews[j] = highQCViews[j], highQCViews[i]
+	})
+
+	// create a block that has the signers in its QC
+	parent := helper.MakeBlock(helper.WithBlockView(view - 1))
+	s.block = helper.MakeBlock(helper.WithBlockView(view),
+		helper.WithParentBlock(parent),
+		helper.WithParentSigners(s.indices))
+	s.tc = helper.MakeTC(helper.WithTCHighestQC(s.block.QC),
+		helper.WithTCView(view+1),
+		helper.WithTCSigners(s.indices),
+		helper.WithTCHighQCViews(highQCViews))
+
+	// return the correct participants and identities from view state
+	s.committee = &mocks.DynamicCommittee{}
+	s.committee.On("IdentitiesByEpoch", mock.Anything, mock.Anything).Return(
+		func(view uint64) flow.IdentityList {
+			return s.participants
+		},
+		nil,
+	)
+	s.committee.On("WeightThresholdForView", mock.Anything).Return(committees.WeightThresholdToBuildQC(s.participants.TotalWeight()), nil)
+
+	s.verifier = &mocks.Verifier{}
+	s.verifier.On("VerifyQC", s.signers, s.block.QC.SigData, parent.View, parent.BlockID).Return(nil)
+
+	// set up the validator with the mocked dependencies
+	s.validator = New(s.committee, s.verifier)
+}
+
+// TestTCOk tests if happy-path returns correct result
+func (s *TCSuite) TestTCOk() {
+	s.verifier.On("VerifyTC", s.signers, []byte(s.tc.SigData), s.tc.View, s.tc.TOHighQCViews).Return(nil).Once()
+
+	// check the default happy case passes
+	err := s.validator.ValidateTC(s.tc)
+	assert.NoError(s.T(), err, "a valid TC should be accepted")
+}
+
+// TestTCEmptySigners tests if correct error is returned when signers are empty
+func (s *TCSuite) TestTCEmptySigners() {
+	s.verifier.On("VerifyTC", mock.Anything, []byte(s.tc.SigData), s.tc.View, s.tc.TOHighQCViews).Return(
+		fmt.Errorf("%w", model.NewInsufficientSignaturesErrorf("")))
```

Comment:
⚠️  `VerifyTC` [does not return `InsufficientSignaturesError`](https://github.com/onflow/flow-go/blob/32eec3b47a466dc14712fa49c19f0ab83966f529/consensus/hotstuff/verification/staking_verifier.go#L120-L145) from what I can see. 

In the case of empty signers, [we will return `InvalidFormatError`](https://github.com/onflow/flow-go/blob/32eec3b47a466dc14712fa49c19f0ab83966f529/consensus/hotstuff/verification/staking_verifier.go#L122), which is considered invalid TC rather than internal error (opposite of what this test states). 

I suggest we modify `ValidateTC` to return consistent errors with `ValidateQC` for this error (eg. both should return `InsufficientSignaturesError` under the same circumstances). We could consider moving this test case to `TestTCInvalidSignature` where we test the other cases of `Verifier` returning various errors. 

Also noticed: 
* For `VerifyQC`, [`CombinedVerifier`](https://github.com/onflow/flow-go/blob/32eec3b47a466dc14712fa49c19f0ab83966f529/consensus/hotstuff/verification/combined_verifier_v2.go#L123) returns `InsufficientSignaturesError` but [`StakingVerifier`](https://github.com/onflow/flow-go/blob/32eec3b47a466dc14712fa49c19f0ab83966f529/consensus/hotstuff/verification/staking_verifier.go#L68) returns `InvalidFormatError`

Reply:
fixed